lib/methods/import.js

```diff
@@ -292,6 +329,144 @@ module.exports = self => {
       };
     },
 
+    async insertOrUpdateDocWithKey(req, {
+      doc,
+      updateKey,
+      updateField
+    }) {
+      const manager = self.apos.doc.getManager(doc.type);
+
+      if (!manager) {
+        await self.apos.notify(req, 'aposImportExport:typeUnknownWithUpdateKey', {
+          interpolate: {
+            type: doc.type,
+            updateKey: doc[updateKey]
+          },
+          dismiss: true,
+          icon: 'database-import-icon',
+          type: 'danger'
+        });
+        throw new Error(`No manager found for this module: ${doc.type}`);
+      }
+
+      if (!self.canImport(req, doc.type)) {
+        throw new Error(`Import is disabled for this module: ${doc.type}`);
+      }
+
+      // `self.apos.page.isPage` does not work here since
+      // the slug can be omitted in a CSV or Excel file.
+      const isPage = manager.__meta.chain.some(module => module.name === '@apostrophecms/page-type');
+
+      if (!doc[updateField]) {
+        doc[updateField] = doc[updateKey];
+      }
+
+      self.handleRichTextFields(manager, doc);
+
+      if (!doc[updateKey]) {
+        return insert();
+      }
+
+      const matchingDraftPromise = manager.findForEditing(
+        req.clone({ mode: 'draft' }),
+        {
+          aposMode: 'draft',
+          [updateField]: doc[updateKey]
+        }).toObject();
+
+      const matchingPublishedPromise = manager.findForEditing(
+        req.clone({ mode: 'published' }),
+        {
+          aposMode: 'published',
+          [updateField]: doc[updateKey]
+        }).toObject();
+
+      let [ matchingDraft, matchingPublished ] = await Promise.all([
+        matchingDraftPromise,
+        matchingPublishedPromise
+      ]);
+
+      if (!matchingDraft && !matchingPublished) {
+        return insert();
+      }
```

Comment:
if neither the draft nor the published version matched the column key value, then we insert the doc.